tools/shared/src/main/scala/org/scalajs/core/tools/javascript/JSDesugaring.scala

```diff
@@ -1454,11 +1497,21 @@ object JSDesugaring {
               js.BinaryOp(JSBinaryOp.!==, newLhs, newRhs)
 
             case String_+ =>
-              if (lhs.tpe == StringType || rhs.tpe == StringType)
-                js.BinaryOp(JSBinaryOp.+, newLhs, newRhs)
-              else
-                js.BinaryOp(JSBinaryOp.+, js.BinaryOp(JSBinaryOp.+,
-                    js.StringLiteral(""), newLhs), newRhs)
+              if (isStrongMode) {
+                val lhsStr =
+                  if (lhs.tpe == StringType) newLhs
+                  else genCallHelper("toString", newLhs)
+                val rhsStr =
+                  if (rhs.tpe == StringType) newRhs
+                  else genCallHelper("toString", newRhs)
```

Comment:
Consider making a helper function for these two vals (which you can use inline below). I think it will make it clearer what it does.